src/detect-engine.c

```diff
@@ -1315,7 +1316,20 @@ bool DetectEngineBufferRunValidateCallback(
 {
     const DetectBufferType *map = DetectEngineBufferTypeGetById(de_ctx, id);
     if (map && map->ValidateCallback) {
-        return map->ValidateCallback(s, sigerror);
```

Comment:
this logic seems to turn a generic validation callback mechanism into a specific one for content. That does not look correct.